src/gui/uithememanager.cpp

```diff
@@ -55,16 +138,56 @@ void UIThemeManager::initInstance()
         m_instance = new UIThemeManager;
 }
 
-UIThemeManager::UIThemeManager()
+UIThemeManager::UIThemeManager() : m_useCustomStylesheet(false)
 {
     const Preferences *const pref = Preferences::instance();
-    if (pref->useCustomUITheme()
-        && !QResource::registerResource(pref->customUIThemePath(), "/uitheme"))
-        LogMsg(tr("Failed to load UI theme from file: \"%1\"").arg(pref->customUIThemePath()), Log::WARNING);
+    bool useCustomUITheme = pref->useCustomUITheme();
 
-#if (defined(Q_OS_UNIX) && !defined(Q_OS_MACOS))
+#ifdef QBT_SYSTEMICONS
     m_useSystemTheme = pref->useSystemIconTheme();
+    m_systemIconMap = systemIconConfig;
+#endif
+
+    QHash<QString, QString> customIconMap;
+
+    if (useCustomUITheme) {
+        try {
+            if (!QResource::registerResource(pref->customUIThemePath(), "/uitheme"))
+                throw ThemeError(tr("Failed to load UI theme from file: \"%1\".")
+                                 .arg(pref->customUIThemePath()));
+            m_useCustomStylesheet = QFile::exists(":uitheme/stylesheet.qss");
+            m_flagsDir = ":uitheme/icons/flags/";
```

Comment:
Is there any real reason to allow custom flags? Can they be different from the default ones?

Reply:
well I have seen people with custom flags but this will be very cumbersome for any one to do. So let me remove them for now.

Reply:
removed custom `m_flagsdir`